connect-inject/handler.go

```diff
@@ -506,22 +394,16 @@ func (h *Handler) shouldInject(pod *corev1.Pod, namespace string) (bool, error)
 	return !h.RequireAnnotation, nil
 }
 
-func (h *Handler) defaultAnnotations(pod *corev1.Pod, patches *[]jsonpatch.JsonPatchOperation) error {
-	if pod.ObjectMeta.Annotations == nil {
-		pod.ObjectMeta.Annotations = make(map[string]string)
+func (h *Handler) defaultAnnotations(pod *corev1.Pod) error {
+	if pod.Annotations == nil {
+		pod.Annotations = make(map[string]string)
 	}
 
 	// Default service name is the name of the first container.
 	if _, ok := pod.ObjectMeta.Annotations[annotationService]; !ok {
 		if cs := pod.Spec.Containers; len(cs) > 0 {
-			// Create the patch for this first, so that the Annotation
-			// object will be created if necessary
-			*patches = append(*patches, updateAnnotation(
-				pod.Annotations,
-				map[string]string{annotationService: cs[0].Name})...)
-
 			// Set the annotation for checking in shouldInject
-			pod.ObjectMeta.Annotations[annotationService] = cs[0].Name
+			pod.Annotations[annotationService] = cs[0].Name
```

Comment:
I was just curious about why we no longer need to use the `pod.ObjectMeta.Annotation` fields, is it because that was used for the old patching and isn't necessary now? I do see that we had sometimes been using `pod.Annotations[]` elsewhere instead of ObjectMeta. 🤷🏽  Non-blocker, just curious.

Reply:
pod.Annotations[] == pod.ObjectMeta.Annotations[] and is supported natively by the kubernetes library and is shorter and im lazy 😄